signerverifier/signerverifier.go

```diff
@@ -0,0 +1,37 @@
+package signerverifier
+
+import (
+	"errors"
+)
+
+var KeyIDHashAlgorithms = []string{"sha256", "sha512"}
+
+var (
+	ErrNotPrivateKey               = errors.New("loaded key is not a private key")
+	ErrSignatureVerificationFailed = errors.New("failed to verify signature")
+	ErrUnknownKeyType              = errors.New("unknown key type")
+	ErrInvalidThreshold            = errors.New("threshold is either less than 1 or greater than number of provided public keys")
+)
+
+const (
+	PublicKeyPEM  = "PUBLIC KEY"
+	PrivateKeyPEM = "PRIVATE KEY"
+)
+
+type SSLibKey struct {
+	KeyIDHashAlgorithms []string `json:"keyid_hash_algorithms"`
+	KeyType             string   `json:"keytype"`
+	KeyVal              KeyVal   `json:"keyval"`
+	Scheme              string   `json:"scheme"`
+	keyID               string
```

Comment:
missing `json:"keyID"`?

Reply:
I'm not sure of the best way to handle KeyID. Make it public and calculate only if it's missing (some files in the custom format don't have it, as it happens).

Reply:
I opted for recalculating it every time

Reply:
Made it public and added the json tag